compute_endpoint/globus_compute_endpoint/engines/helper.py

```diff
@@ -66,6 +70,7 @@ def execute_task(
             error_details=error_details,
         )
 
+    env_details["endpoint_id"] = endpoint_id
```

Comment:
I think this is all fine, just two questions that you probably already thought about:  why add "endpoint_id" value here instead of centralizing it in get_env_details(), and are there any backwards compatibility issues?

ie. for the second, I think the deserialized info is just a modified dict, so no method signatures need to be changed or to account for an extra value, right?

Reply:
> why add "endpoint_id" value here instead of centralizing it in get_env_details()

Since `get_env_details` is called on the SDK side too, where endpoint_id is a meaningless datum

> are there any backwards compatibility issues

Nope, the `details` field is just a straight up `dict` with no impositions on its structure.

A third unasked question is "why do it in `details` instead of higher up in the `Result` structure?" That was in order to avoid needing to modify the web service, and avoid needing to think about backward compatibility in a pydantic-validated model.

Reply:
Makes sense.  (And yes, one of the reasons for adding ``details`` as a dict in the first place is to make it easy to add things to it without having to modifying middlemen code)